### PR TITLE
Initial implementation of Stateful Precompile Contracts

### PR DESCRIPTION
**Pull request by evgeniy-scherbina**
Moved to https://github.com/Kava-Labs/kava/pull/1826

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,6 +14,10 @@ WORKDIR /root/kava
 COPY ./go.mod ./
 COPY ./go.sum ./
 
+RUN echo 3
+RUN mkdir -p /Users/yevheniishcherbina/go/src/github.com/Kava-Labs
```
Comment (galxy25) [inline Dockerfile:18]: Recommend making this a variable vs hard-coded, for example

https://stackoverflow.com/questions/40902445/using-variable-interpolation-in-string-in-docker

Comment (evgeniy-scherbina) [reply Dockerfile:18]: It will be removed, it's temporary

File: docker-build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+rm -rf ./docker-deps && mkdir -p ./docker-deps
+cp -r /Users/yevheniishcherbina/go/src/github.com/Kava-Labs/ethermint docker-deps/
```
Comment (galxy25) [inline docker-build.sh:2]: Recommend making this a variable vs hard-coded, for example

https://stackoverflow.com/questions/40902445/using-variable-interpolation-in-string-in-docker

**Review by galxy25 — APPROVED** [review]

File: docker-build.sh (hunk 2 above)
Comment (evgeniy-scherbina) [reply docker-build.sh]: It will be removed, it's temporary

**Pull request closed without merging**